Compile failure on sparc Solaris 9

Compile fails on Solaris 9 as Solaris 9 does not have `strerror_r`:
```
[solaris_sparc] /home/jenkins/build/workspace/Fractions_XPlatform@2/3rdparty/spdlog_1fcc89/include/spdlog/details/os.h:430:36: error: no matching function for call to ‘errno_to_string(char [256], fmt::internal::Null<>&)’
[solaris_sparc]      return errno_to_string(buf, err);              // use overloading to select correct stringify function
```
this is technically a failure in `fmt` for not falling back to a thread-unsafe version on systems where there is no thread-safe version perhaps?

However, this can be fixed in `spdlog` by adding an inline overload for `::Null<>` - will submit a PR soon.

--- a/include/spdlog/common.h
+++ b/include/spdlog/common.h
@@ -143,11 +143,6 @@ enum class pattern_time_type
 //
 // Log exception
 //
-namespace details {
-namespace os {
-std::string errno_str(int err_num);
-}
-} // namespace details
 class spdlog_ex : public std::exception
 {
 public:
@@ -158,7 +153,15 @@ public:
 
     spdlog_ex(const std::string &msg, int last_errno)
     {
-        _msg = msg + ": " + details::os::errno_str(last_errno);
+        char buf[256], *buf_ptr = buf;
+        SPDLOG_CONSTEXPR auto buf_size = sizeof(buf);
+        if (fmt::safe_strerror(last_errno, buf_ptr, buf_size) != 0)
+        {
+            buf_ptr = buf;
+            char unknown[] = "Unknown error";
+            std::copy(unknown, unknown + sizeof(unknown), buf_ptr);
+        }
+        _msg = msg + ": " + std::string(buf_ptr);
     }
 
     const char *what() const SPDLOG_NOEXCEPT override

--- a/include/spdlog/details/os.h
+++ b/include/spdlog/details/os.h
@@ -383,18 +383,6 @@ inline std::string filename_to_str(const filename_t &filename)
 }
 #endif
 
-// Return errno string (thread safe)
-inline std::string errno_str(int err_num)
-{
-    char buf[256], *buf_ptr = buf;
-    SPDLOG_CONSTEXPR auto buf_size = sizeof(buf);
-    if (fmt::safe_strerror(err_num, buf_ptr, buf_size) == 0)
-    {
-        return std::string(buf_ptr);
-    }
-    return "Unknown error";
-}
-
 inline int pid()
 {
 

--- a/include/spdlog/details/os.h
+++ b/include/spdlog/details/os.h
@@ -383,57 +383,16 @@ inline std::string filename_to_str(const filename_t &filename)
 }
 #endif
 
-inline std::string errno_to_string(char[256], char *res)
-{
-    return std::string(res);
-}
-
-inline std::string errno_to_string(char buf[256], int res)
-{
-    if (res == 0)
-    {
-        return std::string(buf);
-    }
-    return "Unknown error";
-}
-
-inline std::string errno_to_string(char buf[256], const fmt::internal::Null<> &/*tag*/)
-{
-    return errno_to_string(buf, -1);
-}
-
 // Return errno string (thread safe)
 inline std::string errno_str(int err_num)
 {
-    char buf[256];
+    char buf[256], *buf_ptr = buf;
     SPDLOG_CONSTEXPR auto buf_size = sizeof(buf);
-
-#ifdef _WIN32
-    if (strerror_s(buf, buf_size, err_num) == 0)
+    if (fmt::safe_strerror(err_num, buf_ptr, buf_size) == 0)
     {
-        return std::string(buf);
+        return std::string(buf_ptr);
     }
-    else
-    {
-        return "Unknown error";
-    }
-
-#elif defined(__FreeBSD__) || defined(__APPLE__) || defined(ANDROID) || defined(__SUNPRO_CC) ||                                            \
-    ((_POSIX_C_SOURCE >= 200112L) && !defined(_GNU_SOURCE)) // posix version
-
-    if (strerror_r(err_num, buf, buf_size) == 0)
-    {
-        return std::string(buf);
-    }
-    else
-    {
-        return "Unknown error";
-    }
-
-#else // gnu version (might not use the given buf, so its retval pointer must be used)
-    auto err = strerror_r(err_num, buf, buf_size); // let compiler choose type
-    return errno_to_string(buf, err);              // use overloading to select correct stringify function
-#endif
+    return "Unknown error";
 }
 
 inline int pid()

--- a/include/spdlog/details/os.h
+++ b/include/spdlog/details/os.h
@@ -397,6 +397,11 @@ inline std::string errno_to_string(char buf[256], int res)
     return "Unknown error";
 }
 
+inline std::string errno_to_string(char buf[256], const fmt::internal::Null<> &/*tag*/)
+{
+    return errno_to_string(buf, -1);
+}
+
 // Return errno string (thread safe)
 inline std::string errno_str(int err_num)
 {

